GameLogic/engine/IStandardGameLogic.cs

```diff
@@ -4,25 +4,22 @@
 namespace GameLogic
 {
     // The interface represents the logic of chess game. 
-    public interface IStandardGameLogic
+    public interface IStandardGameLogic<T>
```

Comment:
What are potential values for `T` `IStandardGameLogic`? Only `StandardGame`?

I feel like it can become `IGameLogic`.

Reply:
The potential value is class which implements IStandardGameLogic. For example, StandardGame works.
Please refer to https://stackoverflow.com/questions/34881220/c-sharp-interface-method-must-have-return-type-that-derives-from-itself. Could you please suggest your solution? Only generic class worked for me.

P.S. There is no IGameLogic class in project.

Reply:
My question is more about the naming. Why is it `IStandardGameLogic<T>`?

I suggest naming it `IGameLogic<T>` as the interface more or less defines methods necessary for various types of Logics.

For example, later we can have the following implementations:
`class StandardGame : IGameLogic<StandardGame>`
`class FogOfWarGame : IGameLogic<FogOfWarGame>`.


==================================================

But frankly speaking the fact that we have to specify the same class in the generic parameters looks weird. I suggest having this approach for now and changing it later if necessary. We should discuss it in person, it is a tricky topic.

Reply:
I can see that you renamed it. We may consider it resolved.